app/src/main/java/org/oppia/app/player/audio/LanguageDialogFragment.kt

```diff
@@ -0,0 +1,72 @@
+package org.oppia.app.player.audio
+
+import android.app.Dialog
+import android.content.Context
+import android.os.Bundle
+import android.util.Log
+import androidx.appcompat.app.AlertDialog
+import androidx.fragment.app.DialogFragment
+import org.oppia.app.R
+import java.util.ArrayList
+
+private const val KEY_LANGUAGE_LIST = "LANGUAGE_LIST"
+private const val KEY_CURRENT_LANGUAGE = "CURRENT_LANGUAGE"
+
+/**
+ * DialogFragment that controls language selection in audio and written translations
+ */
+class LanguageDialogFragment : DialogFragment() {
+  companion object {
+    /**
+     * This function is responsible for displaying content in DialogFragment
+     * @param languageCodeList: List of strings containing languages
+     * @param currentLanguageCode: Currently selected language code
+     * @return LanguageDialogFragment: DialogFragment
+     */
+    fun newInstance(
+      languageCodeList: List<String>,
+      currentLanguageCode: String
+    ): LanguageDialogFragment {
+      val selectedIndex: Int = languageCodeList.indexOf(currentLanguageCode)
+      val fragment = LanguageDialogFragment()
+      val args = Bundle()
+      args.putStringArrayList(
+        KEY_LANGUAGE_LIST,
+        languageCodeList as ArrayList<String>
+      )
+      args.putInt(KEY_CURRENT_LANGUAGE, selectedIndex)
+      fragment.arguments = args
+      return fragment
+    }
+  }
+
+  override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
+    var languageList = savedInstanceState?.getStringArrayList(KEY_LANGUAGE_LIST)
+    val currentIndex = savedInstanceState?.getInt(KEY_CURRENT_LANGUAGE, 0)
+
+    Log.d("TAG1", "currentIndex " + currentIndex)
```

Comment:
Nit here & elsewhere: you can inline variables in Kotlin strings for convenience, e.g.:

```kotlin
logger.d("TAG", "currentIndex $curentIndex")
```

Also, are these logs needed? The 'TAG1' isn't great if we want check them in. If we do, we should use a more descriptive tag. If not, please remove them. You may also not need to introduce the injectable dialog fragment mentioned above if there's nothing to be injected here.

Reply:
Got it. Will optimise on Logger statements accordingly wherever needed for now I have removed this line because we don't need this.